src/Core/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -0,0 +1,533 @@
+    private $jsonSchemaTypeFactory;
+    private $openApiOptions;
+    private $paginationOptions;
+    private $identifiersExtractor;
+
+    public function __construct(ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, SchemaFactoryInterface $jsonSchemaFactory, TypeFactoryInterface $jsonSchemaTypeFactory, OperationPathResolverInterface $operationPathResolver, ContainerInterface $filterLocator, SubresourceOperationFactoryInterface $subresourceOperationFactory, IdentifiersExtractorInterface $identifiersExtractor = null, array $formats = [], Options $openApiOptions = null, PaginationOptions $paginationOptions = null)
+    {
+        $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
+        $this->jsonSchemaFactory = $jsonSchemaFactory;
+        $this->jsonSchemaTypeFactory = $jsonSchemaTypeFactory;
+        $this->formats = $formats;
+        $this->setFilterLocator($filterLocator, true);
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+        $this->operationPathResolver = $operationPathResolver;
+        $this->subresourceOperationFactory = $subresourceOperationFactory;
+        $this->identifiersExtractor = $identifiersExtractor;
+        $this->openApiOptions = $openApiOptions ?: new Options('API Platform');
+        $this->paginationOptions = $paginationOptions ?: new PaginationOptions();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __invoke(array $context = []): OpenApi
+    {
+        $baseUrl = $context[self::BASE_URL] ?? '/';
+        $contact = null === $this->openApiOptions->getContactUrl() || null === $this->openApiOptions->getContactEmail() ? null : new Model\Contact($this->openApiOptions->getContactName(), $this->openApiOptions->getContactUrl(), $this->openApiOptions->getContactEmail());
+        $license = null === $this->openApiOptions->getLicenseName() ? null : new Model\License($this->openApiOptions->getLicenseName(), $this->openApiOptions->getLicenseUrl());
+        $info = new Model\Info($this->openApiOptions->getTitle(), $this->openApiOptions->getVersion(), trim($this->openApiOptions->getDescription()), $this->openApiOptions->getTermsOfService(), $contact, $license);
+        $servers = '/' === $baseUrl || '' === $baseUrl ? [new Model\Server('/')] : [new Model\Server($baseUrl)];
+        $paths = new Model\Paths();
+        $links = [];
+        $schemas = new \ArrayObject();
+
+        foreach ($this->resourceNameCollectionFactory instanceof LegacyResourceNameCollectionFactoryInterface ? $this->resourceNameCollectionFactory->create(true) : $this->resourceNameCollectionFactory->create() as $resourceClass) {
```

Comment:
Interface `LegacyResourceNameCollectionFactoryInterface` does not exist anywhere in the project. I guess you meant the legacy `ResourceNameCollectionFactoryInterface`, but I can't find it anywhere...

What did you mean here?